collector/lib/NetworkConnection.h

```diff
@@ -347,12 +347,13 @@ size_t Hash(const L4ProtoPortPair& pp);
 
 class ContainerEndpoint {
  public:
-  ContainerEndpoint(std::string container, const Endpoint& endpoint, L4Proto l4proto)
-      : container_(std::move(container)), endpoint_(endpoint), l4proto_(l4proto) {}
+  ContainerEndpoint(std::string container, const Endpoint& endpoint, L4Proto l4proto, int pid = -1)
+      : container_(std::move(container)), endpoint_(endpoint), l4proto_(l4proto), pid_(pid) {}
 
   const std::string& container() const { return container_; }
   const Endpoint& endpoint() const { return endpoint_; }
   const L4Proto l4proto() const { return l4proto_; }
+  const int pid() const { return pid_; }
 
   bool operator==(const ContainerEndpoint& other) const {
```

Comment:
Not absolutely sure about the endpoints, but for the connections, the uniqueness in the big map is determined through operator==()
If we want to be able to store several endpoints having the same port but distinct PIDs, then such endpoints should compare as being different.